### ISSUE TITLE
Kafka configuration fails on Minikube and PKS

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-dataflow#3571]
As a user, when deploying the `master`-based deployment YAMLs for Kafka on Minikube and PKS 1.14, I see timeout errors in the Kafka logs. The apps are unable to connect to the broker, so given the readiness/liveness probe events, the apps continue to restart over and over.

The same files work on GKE, however. 

The recent commit switching to a Kafka image that supports the ability to create topic-partitions is the root-cause. See https://github.com/spring-cloud/spring-cloud-dataflow/commit/8e4eb1463434bb9d2cbda55a37d7f1fce71ba9b4.

**Acceptance:**
- Determine the correct and the latest-stable Kafka image that can work in Minikube, PKS, and GKE
- Review the deployment changes with Kafka Operator team
- Depending on the review, change also the Docker Compose to point to the same image for consistency
- Depending on the review, change also the Helm Chart to point to the same image for consistency
- Update the getting-started guides to reflect the change (console output etc.)

**Comment by chrisjs** [comment]
config changes for review located in my branch: https://github.com/chrisjs/spring-cloud-dataflow/commit/6e5a29cd5498b62e54898d41c29b9527f0f44547

**Comment by sabbyanandan** [comment]
We are reverting to `wurstmeister/kafka` image to unblock the 2.3 RC release. On this image, though, we are still able to over-partition a topic on the initial stream deployment. 

Example:

1) Deploy a stream with two partitions for the `log-sink`.

```
dataflow:>stream create  foo --definition "time | log --spring.cloud.stream.kafka.binder.autoAddPartitions=true --spring.cloud.stream.kafka.binder.minPartitionCount=2"

dataflow:>stream deploy --name foo
```

2) Delete the stream.

3) Redeploy the same stream with two instances for the `log-sink`.

```
dataflow:>stream create  foo --definition "time | log"

dataflow:>stream deploy --name foo --properties "deployer.log.count=2"
```

**Comment by sabbyanandan** [comment]
The corresponding docs change to revert to the older name as the reference for Kafka in K8s: https://github.com/spring-io/dataflow.spring.io/pull/147.

**Comment by chrisjs** [comment]
closing in favor of https://github.com/spring-cloud/spring-cloud-dataflow/issues/3585 which is the desired outcome
